core/domain/state_domain.py

```diff
@@ -916,8 +940,917 @@ def get_all_outcomes(self) -> List[Outcome]:
+                if not multi_item_value and (
+                    rule_spec.rule_type ==
+                    'IsEqualToOrderingWithOneItemAtIncorrectPosition'
+                ):
+                    raise utils.ValidationError(
+                        f'The rule \'{rule_spec_index}\' '
+                        f'of answer group \'{ans_group_index}\' '
+                        f'having rule type - IsEqualToOrderingWith'
+                        f'OneItemAtIncorrectPosition should not '
+                        f'be there when the '
+                        f'multiple items in same position '
+                        f'setting is turned off '
+                        f'in DragAndDropSortInput interaction.'
+                    )
+                # Multiple items cannot be in the same place iff the
+                # setting is turned off.
+                if not multi_item_value:
+                    for ele in rule_spec.inputs['x']:
+                        if len(ele) > 1:
+                            raise utils.ValidationError(
+                                f'The rule \'{rule_spec_index}\' of '
+                                f'answer group \'{ans_group_index}\' '
+                                f'have multiple items at same place '
+                                f'when multiple items in same '
+                                f'position settings is turned off '
+                                f'in DragAndDropSortInput interaction.'
+                            )
+
+                # In `HasElementXBeforeElementY` rule, `X` value
+                # should not be equal to `Y` value.
+                if (
+                    rule_spec.rule_type == 'HasElementXBeforeElementY' and
+                    rule_spec.inputs['x'] == rule_spec.inputs['y']
+                ):
+                    raise utils.ValidationError(
+                        f'The rule \'{rule_spec_index}\' of '
+                        f'answer group \'{ans_group_index}\', '
+                        f'the value 1 and value 2 cannot be '
+                        f'same when rule type is '
+                        f'HasElementXBeforeElementY '
+                        f'of DragAndDropSortInput interaction.'
+                    )
+
+                if rule_spec.rule_type == 'HasElementXAtPositionY':
+                    element = rule_spec.inputs['x']
+                    position = rule_spec.inputs['y']
+                    ele_x_at_y_rules.append(
+                        {'element': element, 'position': position}
+                    )
+
+                if (
+                    rule_spec.rule_type ==
+                    'IsEqualToOrderingWithOneItemAtIncorrectPosition'
+                ):
+                    equal_ordering_one_at_incorec_posn.append(
+                        rule_spec.inputs['x']
+                    )
+
+                if rule_spec.rule_type == 'IsEqualToOrdering':
+                    # `IsEqualToOrdering` rule should not have empty values.
```

Comment:
This seems a bit confusing, the if checks whether `rule_spec.inputs['x']` is empty, for "values" I would expect to check smth like `for input in rule_spec.inputs['x'] if len(input) <= 0`

Reply:
No, actually there were several errors that I received while doing the audit job where the input was completely empty. This could be done when we initialized the drag and drop interaction but leave it as it as without entering anything in GUI. This is fixed in frontend though.

Reply:
Then it should rather be 
```suggestion
                    # `IsEqualToOrdering` rule should not have empty value.
```

right?